buildSrc/src/main/kotlin/NativeUtils.kt

```diff
@@ -1,13 +1,30 @@
 /*
- * Copyright 2014-2021 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ * Copyright 2014-2024 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
  */
 
 import org.gradle.api.*
 
-fun Project.posixTargets(): List<String> = nixTargets() + windowsTargets()
+fun Project.posixTargets(): List<String> = nixTargets() + windowsTargets() + androidNativeTargets()
```

Comment:
Should android native be a part of `nixTargets` as it is Unix-like OS? If so, we could drop almost all duplicating code.

Reply:
In theory - it should be, but on practice because of the issues with commonizer there will be a lot of unresolved declarations in `nix` source-set :(

Reply:
Probably the hierarchy should be similar to `atomicfu`: https://github.com/Kotlin/kotlinx-atomicfu/blob/92ff33b4319970b8ab164c61c8ebfb6a69702955/atomicfu/build.gradle#L154-L172

So:
* posix (= native)
  * nix
    * unix
      * linux
      * apple
    * androidNative
      * androidNative32
      * androidNative64
  * windows

Reply:
Here is PR needed for simple hierarchy changes: https://github.com/ktorio/ktor/pull/4349